### DISCUSSION TITLE
Approaching OpenMage: how to upgrade from Magento and PayPal question

### DISCUSSION THREAD

**Discussion by kamzata** [OpenMage/magento-lts#3569] [Questions & Answers]
I'm using Magento 1.9.4.2 CE on Ubuntu 20.04.

My questions are:

- How can I upgrade to OpenMage latest version? 
- Where can I find some upgrade docs?
- Does PayPal OpenMage module use the new PayPal Rest API V2?
- Will the Magento extensions work flawlessy? 

I'm considering the upgrade to OpenMage mainly because of PayPal v1 does not work well anymore with the Cards.

**Comment by pquerner** [comment]
Hi!

Theres an migration doc here:
https://www.openmage.org/magento-lts/migration-guide.html

Mostly it is recommended to use composer if you are coming from a composerized magento-ce installation previously. Otherwise you can also just take the release files published at github as a drop-in replacement.

Magento 1 Modules should work mostly just fine. There is some edge cases in newer versions of OpenMage (20.x, 21.x) where some breaking changes occur. Please consult the release in that case.

Upgrade should take in a test enviroment and intensive testing in my opinion. Also it depends your overall server structure, when it might make sense to jump from version to version in smaller steps.

Because you are on 1.9.4.2, which wasnt the latest official version of Magento 1.9, using OpenMage 19.4.4 would start you at the official 1.9.4.5 version - so please consult the patches between 1.9.4.2 and 1.9.4.5 if there is something you need to do. (I.e. changes to your templates etc.)

As for Paypal I cannot give you an answer, sorry. Install OpenMage locally and test it with the PayPal Sandbox API.

**Reply by empiricompany** [reply]
Wow, this simple guide is very helpful, and I have never found it on the website. Where is it linked?

**Reply by pquerner** [reply]
On the homepage -> "Easy migration" card -> Click "Get Details".
